main.c

```diff
@@ -74,10 +74,11 @@ static struct tcmur_handler *find_handler_by_subtype(gchar *subtype)
 
 int tcmur_register_handler(struct tcmur_handler *handler)
 {
-	if (handler->handle_cmd &&
-	    (handler->read || handler->write || handler->flush)) {
-		tcmu_err("Skip bad handler: %s\n", handler->name);
-		return -1;
+	if (handler->read || handler->write || handler->flush) {
+		if (handler->handle_cmd && !handler->cmdset) {
```

Comment:
This check just means to stop people using both .handle_cmd(no cmdset) and .{read, write, flush}. Strict checking of all the possible cases for store driver is not so necessary because we examine it by human review beforehand.  Let's keep it simple as started and add more gradients if necessary later.